homeassistant/components/dyson.py

```diff
@@ -69,14 +69,17 @@ def setup(hass, config):
             dyson_device = next((d for d in dyson_devices if
                                  d.serial == device["device_id"]), None)
             if dyson_device:
-                connected = dyson_device.connect(None, device["device_ip"],
-                                                 timeout, retry)
-                if connected:
-                    _LOGGER.info("Connected to device %s", dyson_device)
-                    hass.data[DYSON_DEVICES].append(dyson_device)
-                else:
-                    _LOGGER.warning("Unable to connect to device %s",
-                                    dyson_device)
+                try:
+                    connected = dyson_device.connect(device["device_ip"])
```

Comment:
What's the difference between this part of connection logic and the one at line 92-97? Would it be possible to make a connect function that is called wherever it is needed?